src/NuGet.Core/NuGet.Packaging/PackageArchiveReader.cs

```diff
@@ -228,6 +228,24 @@ public IEnumerable<ZipFilePair> EnumeratePackageEntries(IEnumerable<string> pack
             }
         }
 
+        /// <summary>
+        /// Validate all files in package are not traversed outside of the expected extraction path.
+        /// Eg: file entry like ../../foo.dll can get outside of the expected extraction path.
+        /// </summary>
+        public async Task<bool> ValidatePackageEntriesAsync(CancellationToken token)
+        {
+            var files = await GetFilesAsync(token);
+            // This dumy destionation is used to check if the file in package get outside of the extractionPath
+            var dumyDestination = Path.GetFullPath(Path.Combine(".nuget", "packages"));
```

Comment:
`dummyDestination`

Also, the right-hand side will have unintended results.  I tried this in a console app and got:

    C:\Users\dtivel\Documents\Visual Studio 2017\Projects\ConsoleApp11\ConsoleApp11\bin\Debug\netcoreapp2.0\.nuget\packages

That means that the pattern is:

    <CurrentDirectory>\.nuget\packages

This could more rapidly lead to a `System.IO.PathTooLongException`.

Why not just:

    var destinationRootPath = $@"C:\{Guid.NewGuid().ToString("N")}\";

The path need not be predictable.

Also, shouldn't you ensure the path ends with a slash (per the [MSDN example](https://msdn.microsoft.com/en-us/library/system.io.compression.ziparchive(v=vs.110).aspx#Examples))?

Reply:
just tested $@"C:\{Guid.NewGuid().ToString("N")}\" on mac, Path.GetFullPath doesn't treat C:\ as a root. then it will have unintended results. so I will use NuGetEnvironment.GetFolderPath(NuGetFolderPath.NuGetHome)

Reply:
Could there be user access permission issue with NuGetHome? How about TEMP folder as dummy destination?